command/tls_ca_create.go

```diff
@@ -0,0 +1,162 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/nomad/helper/flags"
+	"github.com/hashicorp/nomad/helper/tlsutil"
+	"github.com/hashicorp/nomad/lib/file"
+	"github.com/posener/complete"
+)
+
+type TLSCACreateCommand struct {
+	Meta
+
+	// days is the number of days the CA will be valid for
+	days int
+
+	// constraint provides a spoecific name constraint to the CA which will then
+	// reject any domains other than this.
+	constraint bool
+
+	// domain is used to provide a custom domain for the CA
+	domain string
+
+	// commonName is used to set a common name for the CA
+	commonName string
+
+	// additioanlNameConstraint provides a list of name constraints to the CA which will then
+	// reject any domains other than these.
+	additionalNameConstraint flags.StringFlag
+}
+
+func (c *TLSCACreateCommand) Help() string {
+	helpText := `
+Usage: nomad tls ca create [options]
+
+Create a new Certificate Authority.
+
+Here are some simple examples, more details can be found below or in 
+the documentation.
+
+Create a new Nomad CA
+
+  $ nomad tls ca create
+  ==> CA Certificate saved to: nomad-agent-ca.pem
+  ==> CA Certificate key saved to: nomad-agent-ca-key.pem
+
+  Create a new Nomad CA with a Custiom Domain
+
+  $ nomad tls ca create -domain foo
+  ==> CA Certificate saved to: foo-agent-ca.pem
+  ==> CA Certificate key saved to: foo-agent-ca-key.pem
+
+
+CA Create Options:
+  -additional-name-constraint
+    Add name constraints for the CA. Results in rejecting certificates
+    for other DNS than specified. Can be used multiple times. Only used 
+    in combination with -name-constraint.
+
+  -common-name
+    Common Name of CA. Defaults to Nomad Agent CA..
+
+  -days
+    Provide number of days the CA is valid for from now on.
+    Defaults to 5 years or 1825 days.
+
+  -domain
+    Domain of nomad cluster. Only used in combination with -name-constraint. 
+    Defaults to nomad.
+
+  -name-constraint
+    Add name constraints for the CA. Results in rejecting
+    certificates for other DNS than specified. If turned on localhost and 
+    -domain will be added to the allowed DNS. If the UI is going to be served 
+    over HTTPS its DNS has to be added with -additional-constraint. It is not
+    possible to add that after the fact! Defaults to false.
```

Comment:
I left some comments on the web documentation about the wording that applies here as well, but we've also got some language here referring to "DNS" when we really mean... "domains" I think?